games/game_bladeandsorcery.py

```diff
@@ -1,18 +1,198 @@
+from pathlib import Path
+import json
+import os
+from typing import List
+
+from PyQt6.QtCore import QDateTime, QDir, QLocale, Qt
+from PyQt6.QtGui import QFont
+
+import mobase
+
+from PyQt6.QtWidgets import QSizePolicy, QVBoxLayout, QFormLayout, QLabel, QStyle
+from ..basic_features.basic_save_game_info import (
+    BasicGameSaveGame,
+    BasicGameSaveGameInfo
+)
+
 from ..basic_game import BasicGame
 
 
-class BaSGame(BasicGame):
+class BaSSaveGame(BasicGameSaveGame):
+    def __init__(self, filepath):
+        super().__init__(filepath)
+        self._filepath = Path(filepath)
+        save = open(self._filepath, "rb")
+        save_data = json.load(save)
+        self._gameMode: str = save_data["gameModeId"]
+        self._gender: str = "Male" if save_data["creatureId"] == "PlayerDefaultMale" else "Female"
+        self._ethnicity: str = save_data["ethnicGroupId"]
+        h, m, s = save_data["playTime"].split(":")
+        self._elapsed: tuple[int, int, float] = (int(h), int(m), float(s))
+        self._created: float = os.path.getctime(filepath)
+        self._modified: float = os.path.getmtime(filepath)
+        save.close()
+
+    def getName(self) -> str:
+        return f"{self.getPlayerSlug()} - {self._gameMode}"
+
+    def getCreationTime(self) -> QDateTime:
+        return QDateTime.fromSecsSinceEpoch(int(self._created))
+
+    def getModifiedTime(self) -> QDateTime:
+        return QDateTime.fromSecsSinceEpoch(int(self._modified))
+
+    def getPlayerSlug(self) -> str:
+        return f"{self._gender} {self._ethnicity}"
+
+    def getElapsed(self) -> str:
+        return f"{self._elapsed[0]} hours, {self._elapsed[1]} minutes, {int(self._elapsed[2])} seconds"
+
+    def getGameMode(self) -> str:
+        return self._gameMode
+
+
+class BaSSaveGameInfoWidget(mobase.ISaveGameInfoWidget):
+    def __init__(self, parent=None):
+        super().__init__(parent)
+        self.resize(400, 125)
+        sizePolicy = QSizePolicy(QSizePolicy.Policy.Preferred, QSizePolicy.Policy.Minimum)
```

Comment:
Maybe a `.ui` would be cleaner here?

Reply:
Maybe but I think we could make a generator that could automatically build a column 0 | 1 label | value pair based on an input map or something rather than making all the labels by hand. Perhaps this can be an alternative to the current QPixmap function.